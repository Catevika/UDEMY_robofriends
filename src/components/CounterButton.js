import React, { PureComponent } from 'react';

export class CounterButton extends PureComponent {
	constructor(props) {
		super(props);
		this.state = {
			count: 0
		};
		this.updateCount = this.updateCount.bind(this);
	}

	shouldComponentUpdate(nextProps, nextState) {
		if (this.state.count !== nextState.count) {
			return true;
		}
		return false;
	}

	updateCount() {
		this.setState(state => {
			return { count: state.count + 1 };
		});
	}

	render() {
		console.log('CounterButton');
		return (
			<button color={this.props.color} onClick={this.updateCount}>
				Count: {this.state.count}{' '}
			</button>
		);
	}
}

export default CounterButton;
