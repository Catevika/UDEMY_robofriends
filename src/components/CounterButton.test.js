import React from 'react';
import CounterButton from './CounterButton';
import { shallow } from 'enzyme'; // mount and render can also be used - see Enzyme documentation

describe('CounterButton component', () => {
	it('expects to render CounterButton component', () => {
		const mockStyle = { color: 'red' };
		expect(
			shallow(<CounterButton style={mockStyle} />).debug()
		).toMatchSnapshot();
	});
	it('correctly increments the counter', () => {
		const mockStyle = { color: 'red' };
		const wrapper = shallow(<CounterButton style={mockStyle} />);

		wrapper.find('[id="counter"]').simulate('click');
		expect(wrapper.state()).toEqual({ count: 1 });
		wrapper.find('[id="counter"]').simulate('click');
		expect(wrapper.state()).toEqual({ count: 2 });
		expect(wrapper.props().style).toEqual({ color: 'red' });
	});
});
