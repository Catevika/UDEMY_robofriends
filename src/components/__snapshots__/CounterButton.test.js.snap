// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterButton component expects to render CounterButton component 1`] = `
"<button id=\\"counter\\" style={{...}} onClick={[Function: bound updateCount]}>
  Count: 
  0
   
</button>"
`;
